src/cairoUtilFuncGen/memory/memoryToCalldata.ts

```diff
@@ -167,4 +172,104 @@ export class MemoryToCallDataGen extends StringIndexedFuncGen {
 
     return funcName;
   }
+
+  private createDynamicArrayCopyFunction(key: string, type: TypeNode): string {
+    const funcName = `wm_to_calldata${this.generatedFunctions.size}`;
+    // Set an empty entry so recursive function generation doesn't clash.
+    this.generatedFunctions.set(key, { name: funcName, code: '' });
+    const implicits =
+      '{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr : felt, warp_memory : DictAccess*}';
+
+    const outputType = CairoType.fromSol(type, this.ast, TypeConversionContext.CallDataRef);
+    assert(outputType instanceof CairoDynArray);
+
+    assert(type instanceof ArrayType);
+    assert(type.size === undefined);
+
+    const elementT = type.elementT;
+    if (elementT instanceof ArrayType) {
+      throw new NotSupportedYetError(
+        `Copying dynamic arrays with element type ${printTypeNode(
+          elementT,
+        )} from memory to calldata is not supported yet`,
+      );
+    }
+    const cairoElementType = CairoType.fromSol(
+      elementT,
+      this.ast,
+      TypeConversionContext.CallDataRef,
+    );
+
+    this.generatedFunctions.set(key, {
+      name: funcName,
+      code: [
+        `func ${funcName}${implicits}(mem_loc: felt) -> (retData: ${outputType.toString()}):`,
+        `    alloc_locals`,
+        `    let (len_256) = wm_read_256(mem_loc)`,
+        `    let (ptr : ${outputType.vPtr.toString()}) = alloc()`,
+        `    let (len_felt) = narrow_safe(len_256)`,
+        `    let (ptr) = ${this.createDynArrayReader(elementT)}(len_256, ptr, mem_loc+ 2)`,
```

Comment:
Why does the dynArrayReader function take len_256 and not just len_felt? I don't see anything in it that requires the length be a Uint256

Reply:
This is going to sound crazy, but what if they have more than the felt limit entries lol. 
I actually saw you do this in storageToMemory so I thought that would be more correct.

Reply:
I used uint256 in storageToMemory because wm_alloc takes a uint256, because array lengths are uint256, but I don't think we need to worry about people returning more than 2^251 elements from a function at a time